### ISSUE TITLE
Pydantic error while importing kserve python SDK for Custom Image Build

### ISSUE THREAD

**Issue by yadavshashank** [kserve/kserve#2368]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
I wrote a kserve script as part of creating a Custom image build pack. [1](https://kserve.github.io/website/0.9/modelserving/v1beta1/custom/custom_model/#create-your-custom-model-server-by-extending-model-class)

Ran the stand-alone script.
```
import kserve
from typing import Dict

class TaggingModel(kserve.Model):
    def __init__(self, name: str):
        super().__init__(name)
        self.name = name
    def predict(self, request: Dict) -> Dict:
        inputs = request.get("text")
        return {"received": inputs}
if __name__ == "__main__":
    model = TaggingModel("tagging-model")
    # model.load()
    kserve.ModelServer(workers=1).start([model])
```

Got the error:
```  File "/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/serve.py", line 7, in <module>
    import kserve
  File "/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/env/lib/python3.9/site-packages/kserve/__init__.py", line 17, in <module>
    from kserve.model_server import ModelServer
  File "/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/env/lib/python3.9/site-packages/kserve/model_server.py", line 27, in <module>
    import kserve.handlers as handlers
  File "/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/env/lib/python3.9/site-packages/kserve/handlers/__init__.py", line 15, in <module>
    from .base import NotFoundHandler  # noqa # pylint: disable=unused-import
  File "/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/env/lib/python3.9/site-packages/kserve/handlers/base.py", line 18, in <module>
    from kserve.model_repository import ModelRepository
  File "/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/env/lib/python3.9/site-packages/kserve/model_repository.py", line 17, in <module>
    from ray.serve.api import RayServeHandle
  File "/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/env/lib/python3.9/site-packages/ray/serve/__init__.py", line 2, in <module>
    from ray.serve.api import (start, get_replica_context, shutdown, ingress,
  File "/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/env/lib/python3.9/site-packages/ray/serve/api.py", line 19, in <module>
    from ray.serve.common import DeploymentInfo, GoalId, ReplicaTag
  File "/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/env/lib/python3.9/site-packages/ray/serve/common.py", line 7, in <module>
    from ray.serve.config import DeploymentConfig, ReplicaConfig
  File "/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/env/lib/python3.9/site-packages/ray/serve/config.py", line 8, in <module>
    from pydantic import (BaseModel, NonNegativeFloat, PositiveFloat,
ImportError: cannot import name 'NonNegativeFloat' from 'pydantic' (/home/shyadav/Documents/Prescient/AI/Experiments/Autotagging Kserve/env/lib/python3.9/site-packages/pydantic/__init__.cpython-39-x86_64-linux-gnu.so)
```

**What did you expect to happen:**
Expected a model server to run on a port.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
Python version is 3.9.
I ran this on my local system as a python script. Tried running it as a docker container, but that generated the same error.

**Environment:**

- Istio Version:
- Knative Version:
- KFServing Version:
- Kubeflow version:
- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):
- kserve python SDk: 0.9.0
- pydantic: 1.7.4

**Comment by andyi2it** [comment]
@yadavshashank 
Can you try using pydantic version 1.8+ or above? The `NonNegativeFloat` seems to be available only after 1.8.

**Comment by yadavshashank** [comment]
Yes, that was the problem. I installed spacy 3.0.2 which changed the pydantic version to 1.7.4. But, ray needed pydantic>=1.8.0. That's why this error occured.
